containers/overview/LinkItem.js

```diff
@@ -1,26 +1,41 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { Link } from 'react-router-dom';
+import { withRouter } from 'react-router-dom';
 import { Tooltip, Icon } from 'react-components';
 import { c } from 'ttag';
 
-const LinkItem = ({ route, text, permission }) => {
+const LinkItem = ({ route, text, permission, history }) => {
+    const handleClick = async (event) => {
+        event.preventDefault();
+
+        history.push(route.pathname);
+
+        // Let the router do the navigation
+        await null;
+
+        const el = document.querySelector(`#${route.hash}`);
+        if (el) {
+            el.scrollIntoView({ behavior: 'smooth', block: 'start' });
+        }
+    };
+
     return (
-        <Link to={route}>
+        <a onClick={handleClick} href={`${route.pathname}#${route.hash}`}>
             <span className="mr0-5">{text}</span>
             {permission ? null : (
                 <Tooltip title={c('Tag').t`Premium feature`}>
                     <Icon name="starfull" fill="attention" />
                 </Tooltip>
             )}
-        </Link>
+        </a>
     );
 };
 
 LinkItem.propTypes = {
-    route: PropTypes.string,
+    route: PropTypes.object,
     permission: PropTypes.bool,
-    text: PropTypes.string
+    text: PropTypes.string,
+    history: PropTypes.object
 };
 
-export default LinkItem;
+export default withRouter(LinkItem);
```

Comment:
Instead of wrapping `LinkItem` with `withRouter`, what do you think about passing `history` by prop from his parent component: `Sections`?

Reply:
It's what I did in the first place. It requires to send it down through 4 components and need a PR on the https://github.com/ProtonMail/proton-mail-settings project so I thought it was a bit painful. But totally possible, as you want.

Reply:
That kind of swingling `history` around is done in many places in `contacts`, so feat not! :sweat_smile: